packages/components/src/custom-select-control-v2/test/index.tsx

```diff
@@ -0,0 +1,176 @@
+			name: 'rose blush',
+		} );
+
+		await user.click( nextSelection );
+
+		expect( nextSelection ).toHaveAttribute( 'aria-selected' );
+
+		expect( currentSelectedItem ).toHaveTextContent(
+			`${ defaultValues.length + 1 } items selected`
+		);
+	} );
+
+	it( 'Should be able to deselect items when provided an array', async () => {
+		const user = userEvent.setup();
+
+		const defaultValues = [
+			'aurora borealis green',
+			'incandescent glow',
+			'key lime green',
+			'rose blush',
+			'ultraviolet morning light',
+		];
+
+		render(
+			<CustomSelect defaultValue={ defaultValues } label="Multi-select">
+				{ defaultValues.map( ( item ) => (
+					<CustomSelectItem key={ item } value={ item }>
+						{ item }
+					</CustomSelectItem>
+				) ) }
+			</CustomSelect>
+		);
+
+		const currentSelectedItem = screen.getByRole( 'combobox', {
+			expanded: false,
+		} );
+
+		expect( currentSelectedItem ).toHaveTextContent(
+			`${ defaultValues.length } items selected`
+		);
+
+		await user.click( currentSelectedItem );
+
+		expect( screen.getByRole( 'listbox' ) ).toHaveAttribute(
+			'aria-multiselectable'
+		);
+
+		defaultValues.map( ( option ) =>
+			expect(
+				screen.getByRole( 'option', { name: option, selected: true } )
+			).toBeVisible()
+		);
+
+		const nextSelection = [
+			'aurora borealis green',
+			'rose blush',
+			'incandescent glow',
+		];
+
+		for ( let i = 0; i < nextSelection.length; i++ ) {
+			await user.click(
+				screen.getByRole( 'option', { name: nextSelection[ i ] } )
+			);
+
+			expect(
+				screen.getByRole( 'option', {
+					name: nextSelection[ i ],
+					selected: false,
+				} )
+			).toBeVisible();
+		}
+
+		expect( currentSelectedItem ).toHaveTextContent(
+			`${ defaultValues.length - nextSelection.length } items selected`
+		);
+	} );
+
+	it( 'Should show rendered content as selected value when using `renderControlledValue`', async () => {
+		const user = userEvent.setup();
+
+		const renderControlledValue = ( value: string | string[] ) => {
+			return (
+				<>
+					<img src={ `${ value }.jpg` } alt={ value as string } />
+					<span>{ value }</span>
+				</>
+			);
+		};
+
+		render(
+			<CustomSelect
+				label="Rendered"
+				renderSelectedValue={ renderControlledValue }
+			>
+				<CustomSelectItem value="april-29">
+					{ renderControlledValue( 'april-29' ) }
+				</CustomSelectItem>
+				<CustomSelectItem value="july-9">
+					{ renderControlledValue( 'july-9' ) }
+				</CustomSelectItem>
+			</CustomSelect>
+		);
+
+		const currentSelectedItem = screen.getByRole( 'combobox', {
+			expanded: false,
+		} );
+
+		expect( currentSelectedItem ).toBeVisible();
+
+		expect( screen.getByRole( 'img', { name: 'april-29' } ) ).toBeVisible();
+
+		expect(
+			screen.queryByRole( 'img', { name: 'july-9' } )
+		).not.toBeInTheDocument();
+
+		await user.click( currentSelectedItem );
+
+		expect( screen.getByRole( 'img', { name: 'july-9' } ) ).toBeVisible();
```

Comment:
The `option` role might be more appropriate here, since we want it to appear specifically as an option, not just whatever image. (The `name` is currently parsed as `july-9 july-9` because of the alt text and the span text.)

```suggestion
		expect( screen.getByRole( 'option', { name: 'july-9' } ) ).toBeVisible();
```

Reply:
I chose the image role to ensure the image was visible and not just the text, to highlight the new feature. But perhaps we could have both?

Reply:
>(The name is currently parsed as july-9 july-9 because of the alt text and the span text.)

Curious what your thoughts are on this, since the assertion you shared will fail. Were you thinking we remove the span? Or use an expression to check the name? 

```tsx
expect(
	screen.getByRole( 'option', { name: /july-9/i } )
).toBeVisible();
```

Reply:
> I chose the image role to ensure the image was visible and not just the text, to highlight the new feature. But perhaps we could have both?

Sure, having both makes sense 👍 Asserting for `img` tests the new feature, while asserting for `option` is more specifically an accessibility check.

> Were you thinking we remove the span? Or use an expression to check the name?

I'd say the awkwardness comes from the fact that we should never write HTML like this, where visible text is immediately preceded by an image with alt text that's identical to the text. (It would be redundant for screen readers.)

```jsx
<img src={ `${ value }.jpg` } alt={ value as string } />
<span>{ value }</span>
```

So yes, I think it's fine to remove the text span. I wouldn't do `name: /july-9/i` though, because the point of this assertion is to check that the _entire text representation_ of the render value we passed is wrapped in an `option`.